src/modules/fancyzones/lib/FancyZones.cpp

```diff
@@ -883,11 +883,27 @@ void FancyZones::MoveSizeStartInternal(HWND window, HMONITOR monitor, POINT cons
     {
         m_zoneWindowMoveSize = iter->second;
         m_zoneWindowMoveSize->MoveSizeEnter(window, m_dragEnabled);
+        if (m_settings->GetSettings().showZonesOnAllMonitors)
+        {
+            for (auto [key_monitor, zoneWindow] : m_zoneWindowMap)
+            {
+                if (zoneWindow && zoneWindow != iter->second)
+                {
+                    zoneWindow->ShowZoneWindow();
+                }
+            }
+        }
     }
     else if (m_zoneWindowMoveSize)
     {
-        m_zoneWindowMoveSize->MoveSizeCancel();
         m_zoneWindowMoveSize = nullptr;
+        for (auto [key_monitor, zoneWindow] : m_zoneWindowMap)
+        {
+            if (zoneWindow)

```

Comment:
Do we need to check zoneWindow every time? We would never add it into the map if creation of `ZoneWinodw` failed (`nullptr` returned). You can check `AddZoneWindow` method of `FancyZones`. I know that there is no harm performing this check (except additional instructions), but in that case we have hundreds of places where we can check the validity of some pointers even though we now that they are fine.

Reply:
I agree with you but I still added the checks just because must of the code base seems to have such redundant checks.